java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/RecordPickerActivity.java

```diff
@@ -0,0 +1,50 @@
+package com.microsoft.hsg.android.hvsample;
+
+import com.microsoft.hsg.android.simplexml.HealthVaultApp;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultClient;
+
+import android.app.Activity;
+import android.app.ListActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+public class RecordPickerActivity extends ListActivity {
+	
+	private HealthVaultClient hvclient;
+	
+	@Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.recordpicker);
+        
+        hvclient = new HealthVaultClient();
+        
+        RecordPickerArrayAdapter adapter = 
+	            new RecordPickerArrayAdapter(this, 
+	            		HealthVaultApp.getInstance().getRecordList(),
+	            		hvclient);
+        
+		setListAdapter(adapter);
+    }
+	
+	@Override
+	protected void onStart() {
+		super.onStart();
+		hvclient.start();
+	}
+	
+	@Override
+	protected void onResume() {
+		super.onResume();
+	}
+
+	@Override
+	protected void onStop() {
+		hvclient.stop();
+		super.onStop();
```

Comment:
If it is critical that we stop the client when the activity is gone, we cannot do it in OnStop due to how the Android activity lifecycle works. There is no guarantee that the system will call onStop, whereas onPause will be called. It may be better suited that we using onResume and onPause to start and stop the client.